Clarified differences between flows and subflows. (#1213)
mulesoft/docs-mule-runtime-3: modules/ROOT/pages/about-flows.adoc

You can connect and trigger their execution with xref:flowref-about.adoc[Flow Reference]
components or by using the DataWeave `lookup` function within expressions and
transformations. The function passes the current event to another flow for

> further event processing. In this way, you can think of a flow as a function call that accepts an event as input and then returns the modified event.

Reviewer: `... you can think of a flow as a function ...` -> `... you can think of a flow-ref as a function ...`

> further event processing. In this way, you can think of a Flow Reference as a function call that accepts an event as input and then returns the modified event.

Reviewer 2: ```suggestion
further event processing. You can think of a Flow Reference as a function call that accepts an event as input and then returns the modified event.
```

initially stopped and start it later through
xref:runtime-manager::flow-management.adoc[Runtime Manager].

> You typically use Flow Reference component or `lookup` function only for

Reviewer: clarify that flows with sources may be calles as well

== Using Subflows

> A subflow is a scope that enables you to group event processors, similar to a flow, but with certain differences and limitations:

Reviewer 2: I suggest italics in one situation only: when introducing/defining a "new" term:
```suggestion
A _subflow_ is a scope that enables you to group event processors in a manner similar to that a flow but with certain differences and limitations:
```

* Subflows do not have event sources.
* Subflows do not have an error handling scope.

> * During design, subflows work as a macro that replaces the Flow Reference components that call the subflow. When you build the application, all Flow Reference components that call a subflow are replaced by the contents of the referenced subflow.

Reviewer 2: ```suggestion
* During design, subflows work as macros that replace the Flow Reference components that call them. 
When you build the application, all Flow Reference components that call a subflow are replaced by the contents of the referenced subflow.
```

> * Referencing a subflow has higher performance than referencing a flow.

Reviewer 2: ```suggestion
* Referencing subflows results in better performance than referencing a flow.
```

> === Considerations

Reviewer 2: I've been taught that "Considerations" is an unnecessarily vague heading that can typically replaced by what we REALLY mean by that broad term: Requirements? Restrictions? Limitations? 
In this particular case, you could probably just add this text after the list in the previous section.

> Because the contents of the subflow replace each Flow Reference component that references the subflow, several instances of the event processors inside the subflow exist in the application at runtime. This behavior can cause issues with event processors that depend on unique IDs or instances to execute like Batch processors.

Reviewer 2: Is it obvious to your audience WHY replacing references to a subflow with its contents results in many event processors existing in the app at runtime, and why that affects processors that depend on unique IDs? Do the several processors that result NOT have unique IDs?
Oh. Nevermind. I see it is explained below.

Reviewer 2: ```suggestion
Because the contents of a subflow replace each Flow Reference component that references that subflow, several instances of the event processors inside that subflow exist in the application at runtime. This behavior can cause issues with event processors that depend on unique IDs or instances to execute, such as batch processors.
```

> For example, configuring a Batch Job inside a subflow causes the application to fail during deployment if the subflow is referenced from more than one Flow Reference component. The application fails to deploy because multiple instances of a Batch Job with the same Job Instance ID exist, which is not allowed.

Reviewer 2: ```suggestion
For example, configuring a batch job inside a subflow causes the application to fail during deployment if the subflow is referenced from more than one Flow Reference component. The application fails to deploy because multiple instances of a batch job with the same job instance ID exist, which is not allowed.
```

== Error Handling

Each flow (but not subflow) can have its own error handling. One reason for

> calling a flow through a Flow Reference component is to separate out the error

Reviewer 2: ```suggestion
calling a flow through a Flow Reference component is to isolate and send the error
```

handling to different contexts. For example, a parent flow that receives web
client requests might define HTTP-related error handling. If the parent flow
then calls a JMS queue for further processing, you can put the JMS event

> processors in a separate child flow and call that flow with a Flow Reference

Reviewer 2: Will audiences be confused by "child flow" vs. "subflow"?

== Using Subflows

A subflow is a scope that enables you to group event processors, similar to a flow, but with certain differences and limitations:

* Subflows do not have event sources.
* Subflows do not have an error handling scope.
* During design, subflows work as a macro that replaces the Flow Reference components that call the subflow. When you build the application, all flow-reference components that call a subflow are replaced by the contents of the referenced subflow.
* Referencing a subflow has higher performance than referencing a flow.

=== Considerations

Because the contents of the sub-flow replace each Flow Reference component that references the subflow, several instances of the event processors inside the subflow exist in the application at runtime. This behavior can cause issues with event processors that depend on unique IDs or instances to execute like Batch processors.

> For example, configuring a Batch Job inside a subflow causes the application to fail during the build phase if the subflow is referenced from more than one Flow Reference component. The application fails to build because multiple instances of a Batch Job with the same Job Instance ID exist, which is not allowed.

Reviewer: what do you mean by `build phase`? should it be initialization?
